daemon/network.go

```diff
@@ -113,7 +114,12 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	}
 
 	nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
-	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
+	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(driverOptions))
+	if len(networkOptions) > 0 {
+		if _, ok := networkOptions[netlabel.Internal]; ok {
+			nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
```

Comment:
why do we need a different map of options for this specific flag? It looks like we could keep only one map if the internal flag was checked and deleted before adding the driver options. Something like:

```go
if _, ok := options[netlabel.Internal]; ok {
  nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
  delete(options, netlabel.Internal)
}
nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
```